src/Dapr.Client/Extensions/ReadOnlyMemoryExtensions.cs

```diff
@@ -0,0 +1,36 @@
+﻿// ------------------------------------------------------------------------
+// Copyright 2025 The Dapr Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ------------------------------------------------------------------------
+
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+
+namespace Dapr.Client;
+
+internal static class ReadOnlyMemoryExtensions
+{
+    public static MemoryStream CreateMemoryStream(this ReadOnlyMemory<byte> memory, bool isReadOnly)
+    {
+        if (memory.IsEmpty)
```

Comment:
So it was this missing check that allowed `TryGetArray()` to hang?  Strange.

Reply:
Yeah, I'm not really sure why that would impact this unless there's some scenario where it the array is empty and it doesn't know how to create the MemoryStream (which doesn't seem right).

Reply:
In fact, we have an awfully timely example - here's a run on a new E2E test I'm building out that uses all the examples as test sources: https://github.com/dapr/dotnet-sdk/actions/runs/12779350972/job/35623810397?pr=1444

It fails with a timeout when taking the overload this fix is intended to address, but again, not entirely sure what the disconnect is between the two and why this works, but that doesn't.